_includes/get_a_quote.html

```diff
@@ -8,180 +8,287 @@ <h1>Get a quote</h1>
                 }
-                setBudgetInContact(val);
-                if(val < 0.01) {
-                    return setEvalText(null, `Input a budget, and we'll work something out!`);
+
+                function sendRequest(data, callbacks) {
+                    var req = new XMLHttpRequest();
+                    req.open("POST", form.action, true);
+                    req.setRequestHeader("Content-Type", "application/json");
+                    req.onreadystatechange = function() {
+                        if(req.readyState != 4) {
+                            return;
+                        }
+                        if(req.status == 200) {
+                            return callbacks.success();
+                        }
+                        console.error("Response from server on request", {
+                            status: req.statusText,
+                            responseType: req.responseType,
+                            response: req.response
+                        });
+                        callbacks.failure();
+                    };
+                    req.send(JSON.stringify(data));
                 }
-                if(val < 10000) {
-                    return setEvalText("bad", `${formatValue(val)} DKK won't get you far at all. Get in touch and we can figure out a realistic budget and plan.`);
+
+                function showSuccess() {
+                    contactFormOverlay.classList.remove("contact-form-overlay-visible");
+                    
+                    setTimeout(function() {
+                        contactFormOverlay.innerText = "Got in touch!";
+                        contactFormOverlay.classList.add("contact-form-overlay-visible");
+
+                        setTimeout(function() {
+                            contactFormOverlay.classList.remove("contact-form-overlay-visible");
+
+                            setTimeout(function() {
+                                contactFormOverlay.classList.add("contact-form-overlay-clickthrough");
+                            }, 500);
+                        }, 1500);
+                    }, 200);
+
+                    deleteTextInField(nameField);
+                    deleteTextInField(emailField);
+                    deleteTextInField(messageField);
+                    deleteTextInField(subjectField, function() {
+                        writeTextInField(subjectField, "Quote request");
+                    });
                 }
-                if(val < 50000) {
-                    return setEvalText("bad", `It's pretty hard to run a project for less that 50.000 DKK&mdash;but you can get in touch and we can talk through a realistic budget.`);
+
+                function showFailure() {
+                    contactFormOverlay.classList.remove("contact-form-overlay-visible");
+                    
+                    setTimeout(function() {
+                        contactFormOverlay.innerText = "Something went wrong - try again!";
```

Comment:
Might want to show if this is one of the fields that are invalid (fx typing invalid email).

Reply:
v2 :smile: